arangod/VocBase/LogicalCollection.cpp

```diff
@@ -938,6 +949,16 @@ arangodb::Result LogicalCollection::properties(velocypack::Slice const& slice, b
   TRI_ASSERT(!isSatellite() || replicationFactor == 0);
   _waitForSync = Helper::getBooleanValue(slice, "waitForSync", _waitForSync);
   _sharding->setWriteConcernAndReplicationFactor(writeConcern, replicationFactor);
+  bool useRevs = Helper::getBooleanValue(slice, StaticStrings::UsesRevisionsAsDocumentIds,
+                                         _usesRevisionsAsDocumentIds);
+  if (!useRevs && _usesRevisionsAsDocumentIds) {
+    return Result(
+        TRI_ERROR_BAD_PARAMETER,
+        "collection uses revisions as document IDs, cannot set this to false");
+  }
+  _usesRevisionsAsDocumentIds = useRevs;
```

Comment:
Is it possible to set the `useRevisionsAsDocumentIds` from the outside, via a call to `db.<collection>.properties({ useRevisionsAsDocumentIds: true })`?
Is this harmful if this is done (un)intentionally on a collection that still uses the old format? Doesn't the collection need to be upgraded first?

Or, put differently: should we expose this in the (also) end-user facing collection properties API?

Reply:
This was a concern for me as well. I added this while I was working on the cluster upgrade procedure, as I didn't see a way around setting the properties in the cluster without using this method to react to plan changes or whatever. I can remove it for now and re-add if it's still necessary later on.